apps/site/assets/ts/stop/components/StopPageRedesign.tsx

```diff
@@ -8,30 +8,68 @@ import StopPageHeaderRedesign from "./StopPageHeaderRedesign";
 import Loading from "../../components/Loading";
 import StopPageDepartures from "./StopPageDepartures";
 import Alerts from "../../components/Alerts";
-import { Route } from "../../__v3api";
+import { Alert, Route } from "../../__v3api";
 import { useSchedulesByStop } from "../../hooks/useSchedules";
-import { useAlertsByStop } from "../../hooks/useAlerts";
 import StopPageOverlay from "./StopPageOverlay";
+import { useAlertsByRoute, useAlertsByStop } from "../../hooks/useAlerts";
+import { isCurrentAlert, routeWideAlerts } from "../../models/alert";
 
 const StopPageRedesign = ({
   stopId
 }: {
   stopId: string;
 }): ReactElement<HTMLElement> => {
+  const [routeIdState, setRouteIdState] = useState<string[]>([]);
+  const [routeState, setRouteState] = useState<Route[]>([]);
+  const [alertState, setAlertState] = useState<Alert[]>([]);
+
   const stop = useStop(stopId);
   const routesWithPolylines = useRoutesByStop(stopId);
   // TODO maybe move to the StopDeparturesPage (or keep it here for loading indicator)
   const schedules = useSchedulesByStop(stopId);
   const alerts = useAlertsByStop(stopId);
+  const alertsForRoutes = useAlertsByRoute(routeIdState);
   //
 
+  useEffect(() => {
+    setRouteState(
+      routesWithPolylines
+        ? routesWithPolylines.map(rwp => omit(rwp, "polylines"))
+        : []
+    );
+  }, [routesWithPolylines]);
```

Comment:
question: Why is this needed in a useEffect hook? When would routesWithPolylines change?